### PR TITLE
Generic path card

### PR DESCRIPTION
**Pull request by rabbitonweb**
Part of https://github.com/quasar-analytics/quasar/issues/1716

This provides generic implementation of `pathCard` that will be available for ALL connectors. It relies on `QueryFile` algebra and calculates cardinality by running `select count(*) from apath` for a given `APath`

### CODE REVIEW

**Comment by rabbitonweb** [comment]
Blocked by https://github.com/quasar-analytics/quasar/pull/2042

**Comment by rabbitonweb** [comment]
@wemrysi this is ready for review

File: connector/src/main/scala/quasar/fs/QueryFile.scala [hunk 1]
```diff
@@ -308,6 +308,12 @@ object QueryFile {
       }
     }
 
+    def first(plan: Fix[LogicalPlan]): ExecM[Option[Data]] = for {
+      h  <- unsafe.eval(plan)
+      vs <- hoistToExec(unsafe.more(h))
+      _  <- toExec(unsafe.close(h))
```
Comment (wemrysi) [inline connector/src/main/scala/quasar/fs/QueryFile.scala]: This won't close the handle in the case of an error from `more`. You need to do something like `unsafe.more(h).ensuring(κ(unsafe.close(h)))`.

**Comment by rabbitonweb** [comment]
@wemrysi done :) thx for help!

**Comment by rabbitonweb** [comment]
Closing because the branch was pushed to upstream

**Pull request closed without merging**